src/app.jsx

```diff
@@ -55,38 +59,48 @@ export const Application = () => {
     useEffect(() => {
         if (path === undefined)
             return;
-        setFiles([]);
-        const currentPath = path.slice(0, pathIndex).join("/");
-        const channel = cockpit.channel({
-            payload: "fslist1",
-            path: `/${currentPath}`,
-            superuser: "try",
-            watch: true,
-        });
 
-        channel.addEventListener("message", (ev, data) => {
-            const item = JSON.parse(data);
-            if (item.event === 'present' && item.path[0] !== ".")
-                setFiles(f => [...f, { name: item.path, type: item.type, modified: item.modified }]);
-            else if (item.event === 'deleted') {
-                const deleted_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                setFiles(f => f.filter((res) => { return res.name !== deleted_name }));
-            } else if (item.event === 'created') {
-                const created_name = item.path.slice(item.path.lastIndexOf("/") + 1);
-                const date_modified = item.tag.slice(item.tag.indexOf("-") + 1, item.tag.lastIndexOf("."));
-                if (created_name[0] !== ".") {
-                    setFiles(f => [...f, { name: created_name, type: item.type, modified: date_modified }]);
+        const getFsList = () => {
+            if (channel.current !== null)
+                channel.current.close();
+
+            const currentPath = path.slice(0, pathIndex).join("/");
+            channel.current = cockpit.channel({
+                payload: "fslist1",
+                path: `/${currentPath}`,
+                superuser: "try",
+                watch: true,
+            });
+
+            const files = [];
+            channel.current.addEventListener("message", (ev, data) => {
+                const item = JSON.parse(data);
+                if (item.event === "present") {
```

Comment:
So are you sure this event will *never* happen for watch events, and *only* for the initial read? Otherwise this would miss a state update for watch events.